programs/pluto/ipsec_doi.c

```diff
@@ -489,10 +485,14 @@ ipsecdoi_replace(struct state *st
     }
     else
     {
+	/* Use the newest Parent SA */
+	struct state *parent = find_phase1_state(c
```

Comment:
I wonder if it worth doing a debug log comparing parent to state_with_serialno(st->st_clonedfrom);
I suggest doing that to show that we are moving the child state from #XX to #YY, and also noting when we actually aren't moving it.

Reply:
Sure, I can do that.

Reply:
Added message and updated all QA